adapters/repos/db/index.go

```diff
@@ -1789,11 +1789,22 @@ func (i *Index) optInitLocalShard(ctx context.Context, class *models.Class, shar
 		return nil
 	}
 
+	disableLazyLoad := i.Config.DisableLazyLoadShards
+
+	// it's assumed that when lazy loading is enabled, all the shards are LazyLoadShard instances
+	// thus even the shard has to be loaded, it may be initialized as a lazy one
 	shard, err := i.initShard(ctx, shardName, class, i.metrics.baseMetrics, disableLazyLoad)
 	if err != nil {
 		return err
 	}
 
+	if mustLoad && !disableLazyLoad {
+		err := shard.(*LazyLoadShard).Load(ctx)
```

Comment:
Can we still add a check for `ok` here and avoid at all cost blind type cast ?

On the same note: do you think you could change the original line where the type cast was failing to also include an `ok` check and avoid a whole panic triggering ?